engine/src/retrier.rs

```diff
@@ -400,11 +400,17 @@ where
 							let half_max = max_sleep_duration(initial_request_timeout, attempt) / 2;
 							let sleep_duration = half_max + rand::thread_rng().gen_range(Duration::default()..half_max);
 
-							let error_message = format!("Retrier {name}: Error for request `{request_log}` with id `{request_id}`, attempt `{attempt}`: {e}. Delaying for {:?}", sleep_duration);
-							if attempt == 0 && !matches!(retry_limit, RetryLimit::Limit(1)) {
-								tracing::warn!(error_message);
-							} else {
-								tracing::error!(error_message);
+							let attempt_number = attempt.saturating_add(1);
```

Comment:
why are we incrementing an attempt number here? If it failed on attempt, 0, why do we log it as a failure for attempt 1? The attempt should only be incremented in one place, makes the code easier to read and reason about.

Reply:
It is much clearer when it says "attempt 2/2" meaning 2 of 2. If we log it from 0, it would say "attempt 1" and you will not know if that is the first or second attempt, at a glance "attempt 1" would seem like the first attempt when in fact it was the second.

Reply:
As I said, we don't want to increment the attempt number in two places - this could be dangerous down the line, and is a little confusing to read in the code. As for making the log messages clearer, we can still do this, perhaps by making it clear that the attempts are 0 indexed... I'm not sure we necessary need to show this "2/2" format. As long as we show the values that are actually used (and not incremented) then it's going to be relatively easy to work out what's happening.